packages/react-components/src/components/AtMentionFlyout.tsx

```diff
@@ -0,0 +1,195 @@
+   */
+  onDismiss?: () => void;
+  /**
+   * Optional callback to render an item of the atMention suggestions list.
+   */
+  onRenderSuggestionItem?: (
+    suggestion: AtMentionSuggestion,
+    onSuggestionSelected?: (suggestion: AtMentionSuggestion) => void
+  ) => JSX.Element;
+}
+
+/**
+ * Options to lookup suggestions in the at mention scenario.
+ *
+ * @beta
+ */
+export interface AtMentionLookupOptions {
+  /**
+   * Optional string to set trigger keyword for mention a specific participant.
+   *
+   * @defaultValue `@`
+   */
+  trigger?: string;
+  /**
+   * Optional callback to fetch a list of at mention suggestions base on the query.
+   */
+  onQueryUpdated?: (query: string) => Promise<AtMentionSuggestion[]>;
+  /**
+   * Optional callback to render an item of the atMention suggestions list.
+   */
+  onRenderSuggestionItem?: (
+    suggestion: AtMentionSuggestion,
+    onSuggestionSelected?: (suggestion: AtMentionSuggestion) => void
+  ) => JSX.Element;
+}
+
+/**
+ * Options to display suggestions in the at mention scenario.
+ *
+ * @beta
+ */
+export interface AtMentionDisplayOptions {
+  /**
+   * Optional callback to override render of an mention in a message thread.
+   */
+  onRenderAtMentionSuggestion?: (suggestion: AtMentionSuggestion) => JSX.Element;
+}
+
+/**
+ * Options to lookup suggestions and display mentions in the at mention scenario.
+ *
+ * @beta
+ */
+export type AtMentionOptions = AtMentionLookupOptions & AtMentionDisplayOptions;
+
+/**
+ * At mention suggestion's state, as reflected in the UI.
+ *
+ * @beta
+ */
+export interface AtMentionSuggestion {
+  /** User ID of a mentioned participant or 'everyone' for an @everyone suggestion */
+  userId: string;
+  /** Type of an at mention suggestion */
+  suggestionType: string;
+  /** Display name of a mentioned participant */
+  displayName: string;
+}
+
+/**
+ * Component to render at mention suggestions.
+ *
+ * @internal
+ */
+export const _AtMentionFlyout = (props: _AtMentionFlyoutProps): JSX.Element => {
+  // Temporary implementation for AtMentionFlyout's position.
+  interface Position {
+    top: number;
+    right: number;
+    bottom: number;
+    left: number;
+  }
+  const { suggestions, title = 'Suggestions', target, onRenderSuggestionItem, onSuggestionSelected } = props;
+  const theme = useTheme();
+  /* @conditional-compile-remove(at-mention) */
+  const ids = useIdentifiers();
+  const localeStrings = useLocale().strings.participantItem;
+
+  const [position, setPosition] = useState<Position>({ top: 0, right: 0, bottom: 0, left: 0 });
+  const [hoveredSuggestion, setHoveredSuggestion] = useState<AtMentionSuggestion | undefined>(undefined);
+
+  // Temporary implementation for AtMentionFlyout's position.
+  useEffect(() => {
+    const rect = target?.current?.getBoundingClientRect();
+    const { top = 0, left = 0, right = 0, bottom = 0, height = 0 } = rect ?? {};
+    const flyoutHeight = 212;
```

Comment:
Looks like a temporary solution, but still would be nice to move those numbers to constants.ts